rollup.config.mjs

```diff
@@ -14,136 +14,117 @@ const is_publish = !!process.env.PUBLISH;
+				return {
+					entryFileNames: (entry) => {
+						if (entry.isEntry) {
+							if (entry.name === 'index') return `index.${ext}`;
+							else if (entry.name === 'ssr') return `ssr.${ext}`;
 
-	{
-		input: `src/runtime/ssr.ts`,
-		output: [
-			{
-				file: `ssr.mjs`,
-				format: 'esm',
-				paths: id => replace_relative_internal_import(id, 'index.mjs')
-			},
-			{
-				file: `ssr.js`,
-				format: 'cjs',
-				paths: id => replace_relative_internal_import(id, 'index.js')
-			}
-		],
-		external,
-		plugins: [ts_plugin]
-	},
+							return `${entry.name}/index.${ext}`;
+						}
+					},
 
-	...fs.readdirSync('src/runtime')
-		.filter(dir => fs.statSync(`src/runtime/${dir}`).isDirectory())
-		.map(dir => ({
-			input: `src/runtime/${dir}/index.ts`,
-			output: [
-				{
-					file: `${dir}/index.mjs`,
-					format: 'esm',
-					paths: id => replace_relative_svelte_imports(id, 'index.mjs')
+					manualChunks: (id) => {
+						if (id.endsWith('internal/index.ts')) {
+							return 'internal';
+						}
+					},
+					format,
+					dir: '.',
+				};
+			}
+		),
+		plugins: [
+			replace({
+				preventAssignment: true,
+				values: {
+					__VERSION__: pkg.version,
 				},
-				{
-					file: `${dir}/index.js`,
-					format: 'cjs',
-					paths: id => replace_relative_svelte_imports(id, 'index.js')
-				}
-			],
-			external,
-			plugins: [
-				replace({
-					__VERSION__: pkg.version
-				}),
-				ts_plugin,
-				{
-					writeBundle(_options, bundle) {
+			}),
+			ts_plugin,
+			{
+				writeBundle(options, bundle) {
+					if (options.format !== 'es') return;
+
+					for (const entry of Object.values(bundle)) {
+						const dir = entry.name;
+						if (!entry.isEntry || !runtime_entrypoints[dir]) continue;
+
 						if (dir === 'internal') {
-							const mod = bundle['index.mjs'];
+							const mod = bundle[`internal/index.mjs`];
 							if (mod) {
-								fs.writeFileSync('src/compiler/compile/internal_exports.ts', `// This file is automatically generated\nexport default new Set(${JSON.stringify(mod.exports)});`);
+								fs.writeFileSync(
+									'src/compiler/compile/internal_exports.ts',
+									`// This file is automatically generated\n` +
+										`export default new Set(${JSON.stringify(mod.exports)});`
+								);
 							}
 						}
 
-						fs.writeFileSync(`${dir}/package.json`, JSON.stringify({
-							main: './index',
-							module: './index.mjs',
-							types: './index.d.ts'
-						}, null, '  '));
+						fs.writeFileSync(
+							`${dir}/package.json`,
+							JSON.stringify(
+								{
+									main: './index.js',
+									module: './index.mjs',
+									types: './index.d.ts',
+								},
+								null,
+								'  '
+							)
+						);
 
-						fs.writeFileSync(`${dir}/index.d.ts`, `export * from '../types/runtime/${dir}/index';`);
+						fs.writeFileSync(
+							`${dir}/index.d.ts`,
+							`export * from '../types/runtime/${dir}/index';`
+						);
 					}
-				}
-			]
-		})),
-
+				},
+			},
+		],
+	},
```

Comment:
we've never had trailing commas before. I guess when we setup prettier it will probably remove them

Reply:
We already have a prettier config in sites repo, we can just remove the source folders from prettierignore

Reply:
We can't introduce prettier just yet, the resulting diff would cause conflicts in a lot of the pulls targeted for v4.